libs/designsystem/modal/src/modal/services/can-dismiss.helper.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { ShowAlertCallback } from '../../modal-wrapper/config';
+import { AlertConfig } from '../alert';
+import { AlertHelper } from './alert.helper';
+
+@Injectable()
+export class CanDismissHelper {
+  constructor(private alertHelper: AlertHelper) {}
+
+  public getCanDismissCallback(callback: ShowAlertCallback) {
+    return async () => {
+      const result = await callback();
+
+      if (typeof result === 'boolean') return result;
+
+      const canCloseModal = await this.showAlert(result);
+      return canCloseModal;
+    };
+  }
+
+  public async showAlert(config: AlertConfig): Promise<boolean> {
+    const alert = await this.alertHelper.showAlert(config);
+    const result = await alert.onDidDismiss;
```

Comment:
Sorry @mark-drastrup I just tested this in the branch deployment, and I see now, that it makes the UX a bit laggy as the modal doesn't close until the close-animation of the alert has played out. Can we change it back to `onWillDismiss `? 🙏🏻 🙈 
```suggestion
    const result = await alert.onWillDismiss; // We use `onWillDismiss` because...
```